src/librustc_ast_passes/ast_validation.rs

```diff
@@ -647,24 +647,61 @@ impl<'a> AstValidator<'a> {
             return;
         }
         // Find all generic argument coming after the first constraint...
-        let mut misplaced_args = Vec::new();
-        let mut first = None;
-        for arg in &data.args {
-            match (arg, first) {
-                (AngleBracketedArg::Arg(a), Some(_)) => misplaced_args.push(a.span()),
-                (AngleBracketedArg::Constraint(c), None) => first = Some(c.span),
-                (AngleBracketedArg::Arg(_), None) | (AngleBracketedArg::Constraint(_), Some(_)) => {
-                }
-            }
-        }
+        let constraint_spans = data
+            .args
+            .iter()
+            .filter_map(|arg| match arg {
+                AngleBracketedArg::Constraint(c) => Some(c.span),
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let arg_spans = data
+            .args
+            .iter()
+            .filter_map(|arg| match arg {
+                AngleBracketedArg::Arg(a) => Some(a.span()),
```

Comment:
Could use `.partition_map` from `Itertools` here (we already use that crate).

Reply:
Can you check the current code to see if you still prefer using `partition_map`? Now I'm separating in _three_ which makes using it less of a win, code clarity-wise.

Reply:
Disregard, I meant on the suggestion code, but this is in the spans.

Reply:
Seems like `partition_map` would still be clearer for the spans?
Also, the comments could use some updating now. :)